lib/scripts.js

```diff
@@ -315,7 +315,8 @@ var scripts = {
       });
     }
     var redlock = new Redlock(queue.clients, queue.redlock);
-    return redlock.lock(job.lockKey(), queue.LOCK_RENEW_TIME);
+    return redlock.lock(job.lockKey(), queue.LOCK_RENEW_TIME)
+    .then(job.queue.client.hset(job.queue.toKey(job.jobId), "lockAcquired", 1));
   },
```

Comment:
the problem here is that getting the lock and setting the lockAcquired property is not an atomic operation anymore...